### PR TITLE
Add topic query to  GET /api/articles endpoint. Add tests for the que…

### PR DESCRIPTION
**Pull request by FreemindNation**
Hi, please take a look at this endpoint and kindly provide some feedback. Thank you.

- add tests for topic query to endpoint
- edit controller
- edit model
- edit error handlers

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -127,6 +127,38 @@ describe('GET /api/articles', () => {
...
+            article_img_url: expect.any(String),
+            comment_count: expect.any(Number)
+          })
+        })
+      })
+    });
+
+    test('400: responds with "Article not found" if query is wrong data type', () => {
+      return request(app)
+      .get('/api/articles?topic=water')
```
Comment (MRombi) [inline __tests__/api.test.js:155]: This would be a 404 because water will possibly be a topic in the future

File: __tests__/api.test.js [hunk 2]
```diff
@@ -127,6 +127,38 @@ describe('GET /api/articles', () => {
...
+            topic: 'cats',
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String),
+            comment_count: expect.any(Number)
+          })
+        })
+      })
+    });
+
```
Comment (MRombi) [inline __tests__/api.test.js:152]: Topic=paper exists but does not have any article associated with it, we should respond with a 200 and return an empty array in this case

**Review by MRombi — COMMENTED** [review]
Please take into consideration my feedback 🙂

File: __tests__/api.test.js (hunk 2 above)
Comment (FreemindNation) [reply __tests__/api.test.js:152]: Thank you Micheal. I will adjust accordingly

**Pull request merged**